src/plugins/projectexplorer/jsonwizard/jsonfieldpage.cpp

```diff
@@ -869,19 +871,21 @@ bool ListField::parseData(const QVariant &data, QString *errorMessage)
                 .arg(type(), name());
         return false;
     }
-    if (value.type() != QVariant::List) {
```

Comment:
All changes to upstream code should be done with separate commits and submitted for upstream review.

Reply:
Done inside PR: https://github.com/sailfishos/sailfish-qtcreator/pull/266

Reply:
@yarfruct Note for your convenience, that I do not require that you create separate PRs here. I am OK with creating more PRs if you really find it more convenient, but I believe it would be less bothering for you having single PR here, with changes clearly separated into commits and submit selected commits for upstream adoption. When accepted by upstream, you would update the PR here with possible changes required by upstream and then the PR would be merged at once.

Reply:
@yarfruct One more note that may not be obvious: your PRs are *not* going to to be delayed until we upgrade to the newer Qt Creator release that will contain your commits accepted by upstream. We can merge your PRs with the original commits immediately when those commits appear in an upstream branch.